submissions/yaripey/memory-pair-game/js/game.js

```diff
@@ -0,0 +1,162 @@
+
+  this.mainContainer = document.querySelector('.container')
+  this.stop = false;
+  this.musicMuted = false;
+}
+
+// Function for the elements array initialisation
+Game.prototype.initiateElements = function () {
+  const preShuffleElements = [];
+  for (let i = 0; i < this.amountOfPairs; i++) {
+    const currentElement = getElementName()
+    const newElem = new Element(currentElement, i)
+    const anotherNewElem = new Element(currentElement, i + this.amountOfPairs)
+    preShuffleElements.push(newElem)
+    preShuffleElements.push(anotherNewElem)
+  }
+  this.elements = shuffleArray(preShuffleElements)
+
+}
+
+// Game initialisation function
+// Also stores the main game logic
+Game.prototype.startGame = function (scenesArray) {
+  this.initiateElements()
+
+  const mainGame = new Scene()
+
+  mainGame.mainBlock = document.createElement('div')
+  mainGame.mainBlock.setAttribute('class', 'game-block')
+  this.elements.forEach(elem => {
+    mainGame.mainBlock.appendChild(elem.makeElement())
+  })
+  mainGame.container.appendChild(mainGame.mainBlock)
+
+  this.mainContainer.addEventListener('click', ({ target }) => {
+    if (this.stop) { return }
+    if (!(target.classList.contains('game') || target.classList.contains('game-block'))) {
+      target = selectProperTarget(target)
+      if (target) {
+        const clickedElement = this.elements.find(elem => {
+          return elem.id === parseInt(target.id)
+        })
+        if (clickedElement.present) {
+          if (this.currentOpened === null) {
+            this.currentOpened = clickedElement
+            clickedElement.peek()
+          } else {
+            if (clickedElement.id !== this.currentOpened.id) {
+              clickedElement.peek()
+              const previouslyOpened = this.currentOpened
+              this.currentOpened = null;
+              if (clickedElement.name === previouslyOpened.name) {
+                previouslyOpened.dissapear()
+                clickedElement.dissapear()
+              } else {
+                clickedElement.peek()
+                previouslyOpened.peek()
+              }
+            }
+          }
+        }
+      }
+      this.checkWin()
+    }
+  })
+
+  this.scenes = scenesArray;
+  this.scenes.push(mainGame)
+  this.scenes.push(congratulations)
+  this.nextScene()
+}
+
+// Creation and start of the background music and mute button
+Game.prototype.startMusic = function () {
+  this.music = document.createElement('audio')
+  this.music.setAttribute('src', 'src/music.mp3')
+  this.music.setAttribute('loop', '')
+  this.music.volume = 0.05
+  document.querySelector('body').appendChild(this.music)
+  this.music.play()
+
+  this.muteButton = document.createElement('button')
+  this.muteButton.classList.add('mute')
+  this.muteButton.addEventListener('click', () => {
+    this.muteButton.classList.toggle('muted')
+    if (this.musicMuted) {
+      this.unmuteMusic()
+    } else {
+      this.muteMusic()
+    }
```

Comment:
How about `this.turnMusic(!this.isMusicMuted)` or just `this.toggleMusicVolume()`?